### PR TITLE
Fix bug in rendering actual PC from NEXT_PC with thumb preffix/suffix instructions

### PR DESCRIPTION
**Pull request by 2over12**
Since we group preffix suffix instructions using the current instruction size to work out the mode was buggy 
https://github.com/lifting-bits/remill/blob/b677387f6f7507193ddd4e3c9a2ec8bc2da94fc0/lib/Arch/Sleigh/Thumb2Arch.cpp#L48

instead this passes down the decoding context to figure out where PC should be

### CODE REVIEW

**Comment by 2over12** [comment]
There's an additional issue here in how we compute the patch expressions for sleigh files. We assume we can compute the address of `inst_next` given the value in PC, but we do it using inst_last, in reality regardless of the instruction size this should be -4 on thumb and -8 on arm which requires determining what mode, this will require another macro

**Comment by 2over12** [comment]
replacing all context reg references with a known constant is too naive, we need to just assign the entry value then lift the register like normal. A better way to do this is to allocate the local context reg as a pointer, store the assumed constant, then lift

**Comment by 2over12** [comment]
Should really integrate TestLifting.cpp from ARM and PPC but Im pressed for time at the moment.

File: lib/Arch/Sleigh/Arch.h [hunk 1]
```diff
@@ -95,13 +95,31 @@ class SingleInstructionSleighContext {
...
+struct ContextRegMappings {
+
+ private:
+  std::unordered_map<std::string, std::string> context_reg_mapping;
+  std::unordered_map<std::string, size_t> vnode_size_mapping;
+
+ public:
+  ContextRegMappings(
+      std::unordered_map<std::string, std::string> context_reg_mapping,
+      std::unordered_map<std::string, size_t> vnode_size_mapping)
```
Comment (Ninja3047) [inline lib/Arch/Sleigh/Arch.h:107]: can you put a comment somewhere that explains what vnode_size_mapping is for? looks like it's set maps to 1 for both ppc and thumb?

**Pull request merged**